### PR TITLE
Benefits of testing blogpost

### PR DESCRIPTION
**Pull request by jacobdadams**
I've pulled together some of the benefits I've found from unit testing. Am I missing anything? Anything you would add?

### CODE REVIEW

**Review by stdavis — DISMISSED** [review]
One more thing that might overlap with number 2. If you write the tests first, you are forced to think through how you want to be able to consume your code before you write it.

**Comment by steveoh** [comment]
This is great. Nice job. 

You mention mocking and patching without mentioning what they are. It might be nice to either remove that to keep it simple or quickly explain what that is. 

One thing I really like about testing is how easy it is to try a lot of inputs/cases without having to doctor data or click through a ui. Aka a fast development loop.

**Comment by jacobdadams** [comment]
I've added your suggestions and made a few rewrites; let me know if I captured what you meant.

If everything looks good to you guys, I'll clean up the dates and switch it from draft.

File: _posts/2022-02-28-python-shorts-unit-testing.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  email: jdadams@utah.gov
+date: 2021-02-28 11:00:00
+categories:
+  - Featured
+  - Developer
+tags:
+  - python
+---
+
+In the realm of programming, "unit testing" is an automated way of testing small, individual "units" of your code to ensure that when given a specific set of inputs they return a specific set of outputs in the way you expected them to.
```
Comment (steveoh) [inline _posts/2022-02-28-python-shorts-unit-testing.md:14]: **thought**: the same inputs same outputs thing is called a [pure function](https://en.wikipedia.org/wiki/Pure_function) and also can relate to [idempotence](https://en.wikipedia.org/wiki/Idempotence)

File: _posts/2022-02-28-python-shorts-unit-testing.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+title: "Python Shorts: The Benefits of Unit Testing"
+author:
+  display_name: Jake Adams
+  email: jdadams@utah.gov
+date: 2021-02-28 11:00:00
```
Comment (steveoh) [inline _posts/2022-02-28-python-shorts-unit-testing.md:6]: **chore**: we'll need to freshen up this date and the file name date when you are ready for a merge

**Review by steveoh — DISMISSED** [review]

File: _posts/2022-02-28-python-shorts-unit-testing.md (hunk 1 above)
Comment (jacobdadams) [reply _posts/2022-02-28-python-shorts-unit-testing.md:14]: Interesting, thanks for the references. 

If I'm understanding it correctly I don't think I mean to refer to completely pure functions; a test could make sure that an instance method sets a property on its object. My thought was that given a defined state and set of inputs, a function or method produces the expected outputs and side effects. That seems similar to pure functions but broader. Am I understanding that right?

**Review by steveoh — APPROVED** [review]

**Pull request merged**